editor/components/inner-blocks/index.js

```diff
@@ -1,50 +1,108 @@
 /**
  * External dependencies
  */
+import { isEqual, pick } from 'lodash';
 import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
 import { withContext } from '@wordpress/components';
 import { withViewportMatch } from '@wordpress/viewport';
-import { compose } from '@wordpress/element';
-import { withSelect } from '@wordpress/data';
+import { Component, compose } from '@wordpress/element';
+import { withSelect, withDispatch } from '@wordpress/data';
+import { synchronizeBlocksWithTemplate } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
+import BlockList from '../block-list';
+import { withBlockEditContext } from '../block-edit/context';
 
-function InnerBlocks( {
-	BlockList,
-	layouts,
-	allowedBlocks,
-	template,
-	isSmallScreen,
-	isSelectedBlockInRoot,
-} ) {
-	const classes = classnames( 'editor-inner-blocks', {
-		'has-overlay': isSmallScreen && ! isSelectedBlockInRoot,
-	} );
-
-	return (
-		<div className={ classes }>
-			<BlockList { ...{ layouts, allowedBlocks, template } } />
-		</div>
-	);
+class InnerBlocks extends Component {
+	componentWillReceiveProps( nextProps ) {
```

Comment:
Can we update this to `componentDidUpdate` at the same time?

Reply:
>Can we update this to componentDidUpdate at the same time?

I went down this path and started to realize there's a fair bit about the behavior of block list settings that needs fixing / improvement. Will make a follow up pull request immediately after this merge.